tests/test_api.py

```diff
@@ -101,10 +104,53 @@ def test_api_extract_info_conda_v2(testing_workdir):
 
 
 def test_api_transmute_tarball_to_conda_v2(testing_workdir):
-    tarfile = os.path.join(data_dir, test_package_name + '.tar.bz2')
-    errors = api.transmute(tarfile, '.conda', testing_workdir)
+    def _walk(path):
+        for entry in os.scandir(path):
+            if entry.is_dir(follow_symlinks=False):
+                yield from _walk(entry.path)
+                continue
+            yield entry
+
+    tar_path = os.path.join(data_dir, test_package_name_2 + '.tar.bz2')
+    errors = api.transmute(tar_path, '.conda', testing_workdir)
+    conda_path = os.path.join(testing_workdir, test_package_name_2 + '.conda')
+    assert not errors
+    assert os.path.isfile(conda_path)
+
+    # Verify original contents were all put in the right place
+    pkg_tarbz2 = tarfile.open(tar_path, mode="r:bz2")
+    info_items = [item for item in pkg_tarbz2.getmembers()
+                  if item.path.startswith("info/")]
+    pkg_items = [item for item in pkg_tarbz2.getmembers()
+                 if not item.path.startswith("info/")]
+
+    errors = []
+    for component, expected in (("info", info_items), ("pkg", pkg_items)):
+        with TemporaryDirectory() as root:
+            api.extract(conda_path, root, components=component)
+
+            contents = {
+                os.path.relpath(entry.path, root): {
+                    "is_symlink": entry.is_symlink(),
+                    "target": os.readlink(entry.path) if entry.is_symlink() else None
+                    }
+                for entry in _walk(root)
+                }
+
+            for e in expected:
+                if e.path not in contents:
+                    errors.append(f"'{e.path}' not found in {component} contents")
+                    continue
+
+                ct = contents.pop(e.path)
+                if e.issym():
+                    if not ct["is_symlink"] or ct["target"] != e.linkname:
+                        errors.append(f"{e.name} -> {e.linkname} incorrect in {component} contents")
+                elif not e.isfile():
+                    raise ValueError(f"unexpected item '{e.path}' in tarball")
```

Comment:
Is this intended to be an exception or should it be appended to `errors`?

Reply:
Exception; AFAIK, conda packages should only contain files and symlinks. So seeing something else implies the test data is invalid.

Reply:
No worries. I've added an inline to comment to clarify that logic.